crates/red_knot_python_semantic/resources/mdtest/annotations/literal_string.md

```diff
@@ -38,8 +38,12 @@ bad_nesting: Literal[LiteralString]  # error: [invalid-type-form]
 ```py
 from typing_extensions import LiteralString
 
-a: LiteralString[str]  # error: [invalid-type-form]
-b: LiteralString["foo"]  # error: [invalid-type-form]
+# error: [invalid-type-form]
+a: LiteralString[str]
+
+# error: [invalid-type-form]
+# error: [unresolved-reference] "Name `foo` used when not defined"
+b: LiteralString["foo"]
```

Comment:
We previously failed to infer types for the inner `"foo"` (type) expression. I believe this has not been caught because no such construct was part of the panic test corpus. We did have a file that has a [PEP-764](https://peps.python.org/pep-0764/) inline TypedDict annotation (`person: TypedDict[{"name": str, age: int}] = …`, which is why it was surfaced here when I added `TypedDict` to the same match case.